hummingbot/connector/exchange/probit/probit_api_order_book_data_source.py

```diff
@@ -0,0 +1,254 @@
+                    if "data" in resp_json:
+                        for market in resp_json["data"]:
+                            if market["market_id"] in trading_pairs:
+                                result[market["market_id"]] = float(market["last"])
+
+        return result
+
+    @staticmethod
+    async def fetch_trading_pairs(domain: str = "com") -> List[str]:
+        async with aiohttp.ClientSession() as client:
+            async with client.get(f"{CONSTANTS.MARKETS_URL.format(domain)}") as response:
+                if response.status == 200:
+                    resp_json: Dict[str, Any] = await response.json()
+                    return [market["id"] for market in resp_json["data"]]
+                return []
+
+    @staticmethod
+    async def get_order_book_data(trading_pair: str, domain: str = "com") -> Dict[str, any]:
+        """
+        Get whole orderbook
+        """
+        async with aiohttp.ClientSession() as client:
+            async with client.get(url=f"{CONSTANTS.ORDER_BOOK_URL.format(domain)}",
+                                  params={"market_id": trading_pair}) as response:
+                if response.status != 200:
+                    raise IOError(
+                        f"Error fetching OrderBook for {trading_pair} at {CONSTANTS.ORDER_BOOK_PATH_URL.format(domain)}. "
+                        f"HTTP {response.status}. Response: {await response.json()}"
+                    )
+                return await response.json()
+
+    async def get_new_order_book(self, trading_pair: str) -> OrderBook:
+        snapshot: Dict[str, Any] = await self.get_order_book_data(trading_pair)
+        snapshot_timestamp: int = int(time.time() * 1e3)
+        snapshot_msg: OrderBookMessage = ProbitOrderBook.snapshot_message_from_exchange(
+            snapshot,
+            snapshot_timestamp,
+            metadata={"trading_pair": trading_pair}
+        )
+        order_book = self.order_book_create_function()
+        bids, asks = probit_utils.convert_snapshot_message_to_order_book_row(snapshot_msg)
+        order_book.apply_snapshot(bids, asks, snapshot_msg.update_id)
+        return order_book
+
+    async def _inner_messages(self,
+                              ws: websockets.WebSocketClientProtocol) -> AsyncIterable[str]:
+        try:
+            while True:
+                msg: str = await asyncio.wait_for(ws.recv(), timeout=self.MESSAGE_TIMEOUT)
+                yield msg
+        except asyncio.TimeoutError:
+            try:
+                pong_waiter = await ws.ping()
+                await asyncio.wait_for(pong_waiter, timeout=self.PING_TIMEOUT)
+            except asyncio.TimeoutError:
+                raise
+        except websockets.exceptions.ConnectionClosed:
+            return
+        finally:
+            await ws.close()
+
+    async def listen_for_order_book_diffs_trades(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+        # TODO: Combine both trades and order_book_diffs
```

Comment:
Yea I was trying to reduce the no. of websockets clients we initialise for this but then realise that the respective messages will need to be appended into the appropriate `Queue`s. Ultimately I chose to not include this since, it will require quite a substantial change to `Exchange` as well as `OrderBookTracker`